File: Problem_4.R
isPalindrome <- function(N)
{
    x = toString(N)
    upper = floor(nchar(x)/2)
    for(i in 1:upper)
    {
        if(substr(x,i,i) != substr(x,nchar(x)-i+1,nchar(x)-i+1)) {return(FALSE)}
    }
    return(TRUE)
}

is3DigitDivisible <- function(N)
{
    if(N < 100^2 || N > 999^2) {return(FALSE)}
    for(i in 100:999)
    {
        if(N %% i == 0)
        {
            if(nchar(toString(N / i)) == 3)
            {
                return(TRUE)
            }
        }
    }
    return(FALSE)
}

ptm <- proc.time()
found = 0
for(i in seq(from=997997,to=10000, by =-11))
{
    if(isPalindrome(i))
    {
        if(is3DigitDivisible(i))
        {
            found = i
            break
        }
    }
}
found
proc.time() - ptm

File: 1-10/Problem_4.py
 #PROBLEM 4
from math import floor
import time
start = time.time()

def IsPalindrome (N):
    for i in range(0,len(N)):
        if N[i] != N[len(N)-1-i]:
            return False
    return True

def Is3DigitDivisible (N):
    if (N < 100 ** 2 | N > 999 ** 2):
        return False
    for i in range(100, 1000):
        if N % i == 0:
            M = (int)(N / i);
            if M >= 100 and M <= 999:
                return True

found = False

for i in range(997997, 10000, -11):
    if(IsPalindrome(str(i))):
        if(Is3DigitDivisible(i)):
            found = True
            result = i
            break

print(result)
end=time.time()
print(end-start)
